src/test/java/seedu/address/logic/commands/BuyCommandTest.java

```diff
@@ -0,0 +1,336 @@
+            new GoodQuantity("10"));
+    private static Good boughtGoodDiffGoodQuantity = new Good(new GoodName("Testing good"),
+            new GoodQuantity("99"));
+    private static Good existingGood = new Good(new GoodName("Testing good"),
+            new GoodQuantity("10"));
+    private static Good buyExistingGoodResultGood = new Good(new GoodName("Testing good"),
+            new GoodQuantity("20"));
+
+    @Test
+    public void constructor_nullSupplier_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new BuyCommand(null));
+    }
+
+    @Test
+    public void equals() {
+        BuyCommand buyCommand = new BuyCommand(boughtGood);
+        BuyCommand buyCommandDiffName = new BuyCommand(boughtGoodDiffGoodName);
+        BuyCommand buyCommandDiffQty = new BuyCommand(boughtGoodDiffGoodQuantity);
+
+        // same object -> returns true
+        assertTrue(buyCommand.equals(buyCommand));
+
+        // same values -> returns true
+        BuyCommand buyCommandCopy = new BuyCommand(boughtGood);
+        assertTrue(buyCommand.equals(buyCommandCopy));
+
+        // different types -> returns false
+        assertFalse(buyCommand.equals(1));
+        assertFalse(buyCommand.equals("string"));
+
+        // null -> returns false
+        assertFalse(buyCommand.equals(null));
+
+        // different GoodQuantity -> returns false
+        assertFalse(buyCommand.equals(buyCommandDiffQty));
+
+        // different GoodName -> returns false
+        assertFalse(buyCommand.equals(buyCommandDiffName));
+    }
+
+    @Test
+    public void execute_buyExistingGood_buySuccessful() throws CommandException {
+        ModelStubWithExistingGood modelStub = new ModelStubWithExistingGood();
+
+        CommandResult commandResult = new BuyCommand(boughtGood)
+                .execute(modelStub);
+
+        String expectedFeedback = String.format(BuyCommand.MESSAGE_SUCCESS,
+                boughtGood.getGoodQuantity().goodQuantity, boughtGood.getGoodName().fullGoodName);
+        assertEquals(expectedFeedback, commandResult.getFeedbackToUser());
+
+        assertEquals(Arrays.asList(buyExistingGoodResultGood), modelStub.inventory);
+    }
+
+    @Test
+    public void execute_buyNewGood_buySuccessful() throws CommandException {
+        ModelStubEmptyInventory modelStub = new ModelStubEmptyInventory();
+
+        CommandResult commandResult = new BuyCommand(boughtGood)
+                .execute(modelStub);
+
+        String expectedFeedback = String.format(BuyCommand.MESSAGE_SUCCESS,
+                boughtGood.getGoodQuantity().goodQuantity, boughtGood.getGoodName().fullGoodName);
+
+        assertEquals(expectedFeedback, commandResult.getFeedbackToUser());
+
+        assertEquals(Arrays.asList(boughtGood), modelStub.inventory);
+    }
+
+    @Test
+    public void execute_buyOverflowInventory_throwsCommandException() {
+        //JD: implement this after you have fixed the BuyCommand overflow bug
```

Comment:
I think you can label this as Todo to make it highlighted?

Reply:
ok, labelled it as todo.